technical-docs/designs/infra/managed-identity.md

```diff
@@ -0,0 +1,251 @@
+
+## 2. Out of Scope
+
+An existing pull request for Bedrock currently exists that enables MSI support
+for aks-gitops modules [#995](https://github.com/microsoft/bedrock/pull/995).
+This design document seeks to solely capture a terraform template and
+complementary test.
+
+The following are not included in this proposal:
+
+- Mocking for Terraform Unit Tests
+- Feature revert and Rollback from failed merges
+- Adjusting Cobalt Test Fixture support for current file organization of
+  Bedrock: i.e.: testing files in respective folders for template environments.
+
+## 3. Design Details
+
+This design seeks to introduce modular testing for terraform known as
+`Test Fixtures` based on best practices initially introduced by
+[Project Cobalt](github.com/microsoft/cobalt). The test fixtures decouples
+terraform commands to respective pipeline templats to be called and dynamically
+populated by a targeted template test.
+
+### 3.1 Embed new Infrastructure DevOps Model Flow - Continuous Integration
+
+Bedrock infrastructure integration tests have problematic gaps that do not
+account for terraform unit testing, state validation to live environments and
+staged release management for Bedrock versioning. In this design we wish to
+provide a single template leveraging MSI that verifies a new Infrastructure
+Testing Workflow that improves on the current Bedrock test harness.
+
+This design is intended to address expected core testing functionality
+including:
+
+- Support deployment of application-hosting infrastructure that will eventually
+  house the actual application service components Capture basic metrics and
+  telemetry from the deployment process for monitoring of ongoing pipeline
+  performance and diagnosis of any deployment failures
+- Support deployment into multiple staging environments
+- Execute automated unit-level and integration-level tests against the
+  resources, prior to deployment into any long-living environments
+- Provide a manual approval process to gate deployment into long-living
+  environments
+- Provide detection, abort, and reporting of deployment status when a failure
+  occurs.
+
+![](infratestflow.png)
+
+The proposed new Infrastructure Devops Flow for Terraform Testing can be
+separated by 4 key functionalities:
+
+1. Test Suite Initialization - Provisioning global artifacts, secrets and
+   dependencies needed for targeted whitelisted test matrix.
+2. Static Validation - Environment initialization, code validation, inspection,
+   terraform security compliance, and terraform module unit tests.
+3. Dynamic Validation - Targeted environment interoperability, integration
+   tests, cloud deployment, de-provisioning of resources, error reporting.
+4. QA- Peer approval, release management, feature staging, acceptance test
+   within live cluster.
+
+> The diagram above contains green check marks that indicate preexisting Bedrock
```

Comment:
I'm not able to see the diagram.

Reply:
This is referencing line 81 `infratestflow.png`

Reply:
Security compliance... do we have a set of goals this will target?

Reply:
@jmspring  see: https://github.com/CatalystCode/spk/pull/481#discussion_r404572981